app/controllers/DeclarationController.scala

```diff
@@ -49,15 +52,35 @@ class DeclarationController @Inject()(
   }
 
   def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData).async { implicit request =>
-    val answers: UserAnswers = request.userAnswers.get
-
-    service
-      .createCase(mapper.map(answers))
-      .map(c => answers.set(ConfirmationPage, Confirmation(c.reference)))
-      .flatMap(updatedAnswers => {
-        dataCacheConnector.save(updatedAnswers.cacheMap)
-          .map(_ => Redirect(navigator.nextPage(ConfirmationPage, mode)(updatedAnswers)))
-      })
+
+    val userAnswers: UserAnswers = request.userAnswers.get // TODO: we should not call `get` on an Option
+
+    val newCaseRequest: NewCaseRequest = mapper.map(userAnswers)
+    auditService.auditBTIApplicationSubmission(newCaseRequest)
+
+    service.createCase(newCaseRequest)
+      .map(Some(_))
+      .recover(handleFailingSubmission(newCaseRequest))
+      .flatMap {
+        case Some(c: Case) =>
+          auditService.auditBTIApplicationSubmissionSuccessful(c)
+          userAnswers.set(ConfirmationPage, Confirmation(c.reference))
+
+          dataCacheConnector.save(userAnswers.cacheMap) map ( _ => redirect(mode, userAnswers) )
+        case _ =>
+          Future.successful(redirect(mode, userAnswers))
```

Comment:
If the case create fails do we actually want to redirect to the confirmation page?
I would have thought it would be better to go to an error page, or back to the declaration page so they can try again